framework/src/main/java/org/checkerframework/common/accumulation/AccumulationChecker.java

```diff
@@ -10,20 +13,71 @@
  * An accumulation checker is one that accumulates some property: method calls, map keys, etc.
  *
  * <p>This class provides a basic accumulation analysis that can be extended to implement an
- * accumulation type system. This accumulation analysis represents all facts as Strings. It
- * automatically includes returns-receiver aliasing to precisely handle fluent APIs, but otherwise
- * uses no alias analysis.
+ * accumulation type system. This accumulation analysis represents all facts as Strings.
+ *
+ * <p>This class supports modular alias analyses. To choose the alias analyses that your
+ * accumulation checker uses, override the {@link #createAliasAnalyses()} method. By default, the
+ * only alias analysis used is Returns Receiver.
  *
  * <p>The primary extension point is the constructor of {@link AccumulationAnnotatedTypeFactory},
  * which every subclass should override to provide custom annotations.
  */
 public abstract class AccumulationChecker extends BaseTypeChecker {
 
+    /**
+     * Set of alias analyses that are enabled in this particular accumulation checker. Set to
+     * non-null the first time {@link #isEnabled(AliasAnalysis)} is called, to allow the
+     * implementation of {@link #createAliasAnalyses()} to be as simple as possible.
+     */
+    private @MonotonicNonNull EnumSet<AliasAnalysis> aliasAnalyses;
+
     @Override
     protected LinkedHashSet<Class<? extends BaseTypeChecker>> getImmediateSubcheckerClasses() {
         LinkedHashSet<Class<? extends BaseTypeChecker>> checkers =
                 super.getImmediateSubcheckerClasses();
-        checkers.add(ReturnsReceiverChecker.class);
+        AliasAnalysis[] aliasAnalyses = createAliasAnalyses();
```

Comment:
This is coded as if enabling any `AliasAnalysis` will require enabling a sub-checker.  Is that the intention?  I'm not really sure how future alias reasoning might look.  In any case, this code might be clearer for now by just checking if `aliasAnalyses` contains `RETURNS_RECEIVER`, rather than looping and doing a switch.

Reply:
That's a fair point. I'll change this.